### PR TITLE
Do feat spaces - Update cascade remove in entity space and persist in entity member

### PR DESCRIPTION
**Pull request by NhatDuy98**

### CODE REVIEW

File: src/main/java/org/design_manager_project/dtos/space/SpaceDTO.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class SpaceDTO extends BaseDTO<UUID> {
+
+    @NotBlank(message = "name not blank")
```
Comment (trisjr-old) [inline src/main/java/org/design_manager_project/dtos/space/SpaceDTO.java]: Please refactor this error message

File: src/main/java/org/design_manager_project/services/BaseService.java [hunk 2]
```diff
@@ -38,14 +38,18 @@ public Page<DTO> findAllWithPage(FT ft){
...
+    public Optional<DTO> findById(ID id) {
         E e = baseRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Not found entity with id: " + id));
 
-        Field deletedAtField = e.getClass().getDeclaredField("deletedAt");
-        deletedAtField.setAccessible(true);
-        Object deletedAtValue = deletedAtField.get(e);
-        if (deletedAtValue != null) {
-            throw new BadRequestException(Constants.OBJECT_DELETED);
+        try {
+            Field deletedAtField = e.getClass().getDeclaredField("deletedAt");
```
Comment (trisjr-old) [inline src/main/java/org/design_manager_project/services/BaseService.java:43]: It is possible to create a separate reusable function to check if a record has been deleted. For example, when updating or deleting a record that has already been deleted previously.

**Comment by KreltTruong** [comment]
Approved!!

**Review by trisjr-old — CHANGES_REQUESTED** [review]

File: src/main/java/org/design_manager_project/controllers/SpaceController.java [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+public class SpaceController extends BaseController<Space, SpaceDTO, SpaceFilter, UUID>{
+    private final SpaceService spaceService;
+    protected SpaceController(SpaceService spaceService) {
+        super(spaceService);
+        this.spaceService = spaceService;
+    }
+
+    @GetMapping("/users/{id}")
+    public ApiResponse getSpaceWithUserId(@PathVariable("id") UUID uuid){
+        return ApiResponse.success(spaceService.getSpaceWithUserId(uuid));
```
Comment (trisjr-old) [inline src/main/java/org/design_manager_project/controllers/SpaceController.java]: According to the current logic, it seems possible to create multiple Spaces with the same userId, so the API 'api/spaces/users/{id}' may not be correct? Please confirm this for me.

**Pull request merged**